[Bug]: test_node_affinity_to_ceph_toolbox_pod Failed with error namespaces "openshift-storage-client" not found

### Description
test_node_affinity_to_ceph_toolbox_pod Failed with error namespaces "openshift-storage-client" not found
Need to ajust test case for multiclient setup
### Steps to Reproduce
1.https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/852/41066/1907690/1907844/log
2.
3.

### Actual Behavior
 test_node_affinity_to_ceph_toolbox_pod Failed
### Expected Behavior
 test_node_affinity_to_ceph_toolbox_pod should Pass
### Impact (likelihood of reproduction, impact on the cluster and on other tests, etc.)

### Screenshots (if applicable)

### Environment
- Test Suite(s):
- Platform(s):4.19
- Version(s):
- OS:

### Additional Context

self = <test_cephtoolbox_pod_nodeaffinity.TestCephtoolboxPod object at 0x7f7b96fa89d0>


@tier1
@polarion_id("OCS-6086")
def test_node_affinity_to_ceph_toolbox_pod(self):
    """
    This test verifies whether ceph toolbox failovered or not after applying node affinity

    """
    other_nodes = get_worker_node_where_ceph_toolbox_not_running()
    other_node_name = random.choice(other_nodes)
    log.info(
        "Apply node affinity with a node name other than currently running node."
    )

  assert apply_node_affinity_for_ceph_toolbox(
        other_node_name
    ), "Failed to apply node affinity for the Ceph toolbox on the specified node."

tests/functional/pod_and_daemons/test_cephtoolbox_pod_nodeaffinity.py:68: 



ocs_ci/ocs/node.py:3261: in apply_node_affinity_for_ceph_toolbox
    storagecluster_obj.patch(params=param, format_type="merge")
ocs_ci/ocs/ocp.py:532: in patch
    result = self.exec_oc_cmd(command)
ocs_ci/ocs/ocp.py:216: in exec_oc_cmd
    out = run_cmd(
ocs_ci/utility/utils.py:490: in run_cmd
    completed_process = exec_cmd(
ocs_ci/utility/retry.py:79: in f_retry
    return f(args, *kwargs)



cmd = ['oc', '--kubeconfig', '/home/jenkins/clusters/j085-pcbaie-t1/openshift-cluster-dir/auth/kubeconfig', '-n', 'openshift-storage-client', 'patch', ...]
secrets = None, timeout = 600, ignore_error = False, threading_lock = None
silent = False, use_shell = False
cluster_config = <ocs_ci.framework.MultiClusterConfig object at 0x7f7bc9900150>
lock_timeout = 7200, output_file = None, kwargs = {}
_env = {'BASH_FUNC_which%%': '() {  ( alias;\n eval ${which_declare} ) | /usr/bin/which --tty-only --read-alias --read-functi...ISPLAY_NAME': '#85 - Prod ODF Provider Client - BM AI Encryption - tier1', 'BUILD_ID': '85', 'BUILD_NUMBER': '85', ...}
kubeconfig_path = '/home/jenkins/clusters/j085-c1baie-t1/openshift-cluster-dir/auth/kubeconfig'


@retry(
    CommandFailed,
    tries=6,
    delay=10,
    backoff=1,
    text_in_exception="client connection lost",
)
def exec_cmd(
    cmd,
    secrets=None,
    timeout=600,
    ignore_error=False,
    threading_lock=None,
    silent=False,
    use_shell=False,
    cluster_config=None,
    lock_timeout=7200,
    output_file=None,
    **kwargs,
):
    """
    Run an arbitrary command locally

    If the command is grep and matching pattern is not found, then this function
    returns "command terminated with exit code 1" in stderr.

    Args:
        cmd (str): command to run
        secrets (list): A list of secrets to be masked with asterisks
            This kwarg is popped in order to not interfere with
            subprocess.run(``**kwargs``)
        timeout (int): Timeout for the command, defaults to 600 seconds.
        ignore_error (bool): True if ignore non zero return code and do not
            raise the exception.
        threading_lock (threading.RLock): threading.RLock object that is used
            for handling concurrent oc commands
        silent (bool): If True will silent errors from the server, default false
        use_shell (bool): If True will pass the cmd without splitting
        cluster_config (MultiClusterConfig): In case of multicluster environment this object
                will be non-null
        lock_timeout (int): maximum timeout to wait for lock to prevent deadlocks (default 2 hours)
        output_file (str): path where to write output of stderr from command - apply only when silent mode is True

    Raises:
        CommandFailed: In case the command execution fails

    Returns:
        (CompletedProcess) A CompletedProcess object of the command that was executed
        CompletedProcess attributes:
        args: The list or str args passed to run().
        returncode (str): The exit code of the process, negative for signals.
        stdout     (str): The standard output (None if not captured).
        stderr     (str): The standard error (None if not captured).

    """
    _env = kwargs.pop("env", os.environ.copy())
    kubeconfig_path = config.RUN.get("kubeconfig")
    if kubeconfig_path:
        _env["KUBECONFIG"] = kubeconfig_path
    if cluster_config:
        kubeconfig_path = cluster_config.RUN.get("kubeconfig")
        if kubeconfig_path:
            _env["KUBECONFIG"] = cluster_config.RUN.get("kubeconfig")
    if isinstance(cmd, str) and not kwargs.get("shell"):
        cmd = shlex.split(cmd)
    if (
        kubeconfig_path
        and cmd[0] == "oc"
        and "--kubeconfig" not in cmd
        and "mirror" not in cmd
    ):
        kube_index = 1
        # check if we have an oc plugin in the command
        plugin_list = "oc plugin list"
        cp = subprocess.run(
            shlex.split(plugin_list),
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
        )
        subcmd = cmd[1].split("-")
        if len(subcmd) > 1:
            subcmd = "_".join(subcmd)
        if not isinstance(subcmd, str) and isinstance(subcmd, list):
            subcmd = str(subcmd[0])

        for l in cp.stdout.decode().splitlines():
            if subcmd in l:
                # If oc cmdline has plugin name then we need to push the
                # --kubeconfig to next index
                kube_index = 2
                log.info(f"Found oc plugin {subcmd}")
        cmd = list_insert_at_position(cmd, kube_index, ["--kubeconfig"])
        cmd = list_insert_at_position(cmd, kube_index + 1, [kubeconfig_path])
    try:
        if kwargs.get("shell"):
            masked_cmd = mask_secrets(cmd, secrets)
        else:
            masked_cmd = shlex.join(mask_secrets(cmd, secrets))
        log.info(f"Executing command: {masked_cmd}")
        if threading_lock and cmd[0] == "oc":
            threading_lock.acquire(timeout=lock_timeout)
        completed_process = subprocess.run(
            cmd,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            stdin=subprocess.PIPE,
            timeout=timeout,
            env=_env,
            **kwargs,
        )
    finally:
        if threading_lock and cmd[0] == "oc":
            threading_lock.release()
    masked_stdout = mask_secrets(completed_process.stdout.decode(), secrets)
    if len(completed_process.stdout) > 0:
        log.debug(f"Command stdout: {masked_stdout}")
    else:
        log.debug("Command stdout is empty")

    masked_stderr = mask_secrets(completed_process.stderr.decode(), secrets)
    if len(completed_process.stderr) > 0:
        if not silent:
            log.warning(f"Command stderr: {masked_stderr}")
        else:
            if output_file:
                with open(output_file, "a") as out_fd:
                    out_fd.write(masked_stderr)
    else:
        if not silent:
            log.debug("Command stderr is empty")
    log.debug(f"Command return code: {completed_process.returncode}")
    if completed_process.returncode and not ignore_error:
        masked_stderr = bin_xml_escape(filter_out_emojis(masked_stderr))
        if (
            "grep" in masked_cmd
            and b"command terminated with exit code 1" in completed_process.stderr
        ):
            log.info(f"No results found for grep command: {masked_cmd}")
        else:

          raise CommandFailed(
                f"Error during execution of command: {masked_cmd}."
                f"\nError is {masked_stderr}"
            )
E               ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc --kubeconfig /home/jenkins/clusters/j085-pcbaie-t1/openshift-cluster-dir/auth/kubeconfig -n openshift-storage-client patch storagecluster ocs-storagecluster -n openshift-storage-client -p '{"spec": {"placement": {"toolbox": {"nodeAffinity": {"requiredDuringSchedulingIgnoredDuringExecution": {"nodeSelectorTerms": [{"matchExpressions": [{"key": "kubernetes.io/hostname","operator": "In","values": ["j085-c1baie-t1-8fdn5-wc85f"]}]}]}}}}}}' --type merge.
E               Error is Error from server (NotFound): namespaces "openshift-storage-client" not found



ocs_ci/utility/utils.py:737: CommandFailed

--- a/tests/functional/pod_and_daemons/test_cephtoolbox_pod_nodeaffinity.py
+++ b/tests/functional/pod_and_daemons/test_cephtoolbox_pod_nodeaffinity.py
@@ -5,7 +5,12 @@ import time
 
 from ocs_ci.framework import config
 from ocs_ci.framework.pytest_customization.marks import brown_squad
-from ocs_ci.framework.testlib import tier1, tier4b, polarion_id
+from ocs_ci.framework.testlib import (
+    tier1,
+    tier4b,
+    polarion_id,
+    runs_on_provider,
+)
 from ocs_ci.ocs import ocp, constants
 from ocs_ci.ocs.node import (
     apply_node_affinity_for_ceph_toolbox,
@@ -25,6 +30,7 @@ log = logging.getLogger(__name__)
 
 
 @brown_squad
+@runs_on_provider
 class TestCephtoolboxPod:
     @pytest.fixture(scope="class", autouse=True)
     def teardown(self, request):
@@ -64,6 +70,7 @@ class TestCephtoolboxPod:
     def test_node_affinity_to_ceph_toolbox_pod(self):
         """
         This test verifies whether ceph toolbox failovered or not after applying node affinity
+        In provider-client mode, this runs in the provider cluster where toolbox exists
 
         """
         other_nodes = get_worker_node_where_ceph_toolbox_not_running()
@@ -82,6 +89,7 @@ class TestCephtoolboxPod:
         This test verifies ceph toolbox runs only on the node given in node-affinity.
         Reboot the node after applying node-affinity.
         Expectation is the pod should come up only on that node mentioned in affinity.
+        In provider-client mode, this runs in the provider cluster where toolbox exists
 
         """
         other_nodes = get_worker_node_where_ceph_toolbox_not_running()
@@ -110,7 +118,8 @@ class TestCephtoolboxPod:
     @polarion_id("OCS-6090")
     def test_nodeaffinity_to_ceph_toolbox_with_default_taints(self):
         """
-        This test verifies whether ceph toolbox failovered or not after applying node affinity on tainted node
+        This test verifies whether ceph toolbox failovered or not after applying node affinity on tainted node.
+        In provider-client mode, this runs in the provider cluster where toolbox exists.
 
         """
         worker_nodes = get_worker_nodes()

